tools/gen_addon_readme.py

```diff
@@ -295,14 +295,13 @@ def gen_addon_readme(
         addons.extend(find_addons(addons_dir))
     for addon_dir in addon_dirs:
         addon_name = os.path.basename(os.path.abspath(addon_dir))
-        manifest = read_manifest(addon_dir)
-        addons.append((addon_name, addon_dir, manifest))
-    readme_filenames = []
-    for addon_name, addon_dir, manifest in addons:
         try:
             manifest = read_manifest(addon_dir)
         except NoManifestFound:
             continue
```

Comment:
You must put something like `manifest = False` and check that inside the loop for doing the continue

Reply:
But that's something silly. Sooner or later it's filtered. Why filter it later instead of sooner? Furthermore, if done later, you are repeating the manifest reading for the addons introduced with addons_dir (directory) parameter.

Reply:
Well, it's not silly as beginning with, you will have an error after when adding the tuple. If the continue can be skipped inside the for is a different question.